its/src/test/java/com/sonar/it/scanner/msbuild/sonarcloud/IncrementalPRAnalysisSonarCloudTest.java

```diff
@@ -0,0 +1,221 @@
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+public class IncrementalPRAnalysisSonarCloudTest {
+  private final static Logger LOG = LoggerFactory.getLogger(IncrementalPRAnalysisSonarCloudTest.class);
+  private final static Integer COMMAND_TIMEOUT = 2 * 60 * 1000;
+  private final static String SCANNER_PATH = System.getenv("SCANNER_PATH") == null
+    ? "../build/sonarscanner-msbuild-net46/SonarScanner.MSBuild.exe" // On the local machine, the scanner is prepared by ci-build.ps1 script.
+    : System.getenv("SCANNER_PATH");
+  private final static String[] prArguments = {
+    "/d:sonar.pullrequest.base=master",
+    "/d:sonar.pullrequest.branch=pull-request-branch",
+    "/d:sonar.pullrequest.key=pull-request-key"
+  };
+  private final static String SONARCLOUD_ORGANIZATION = System.getenv("SONARCLOUD_ORGANIZATION");
+  private final static String SONARCLOUD_PROJECT_KEY = System.getenv("SONARCLOUD_PROJECT_KEY");
+  private final static String SONARCLOUD_URL = System.getenv("SONARCLOUD_URL");
+  private final static String SONARCLOUD_PROJECT_TOKEN = System.getenv("SONARCLOUD_PROJECT_TOKEN");
+
+  @ClassRule
+  public static TemporaryFolder temp = TestUtils.createTempFolder();
+
+  @Test
+  public void incrementalPrAnalysis_master() throws IOException {
+    var projectDir = TestUtils.projectDir(temp, "IncrementalPRAnalysis");
+    var logWriter = new StringWriter();
+    StreamConsumer.Pipe logsConsumer = new StreamConsumer.Pipe(logWriter);
+
+    runBeginStep(projectDir, logsConsumer);
+
+    assertThat(logWriter.toString()).contains(
+      "Processing analysis cache",
+      "Incremental PR analysis: Base branch parameter was not provided.",
+      "Cache data is empty. A full analysis will be performed.");
+  }
+
+  @Test
+  public void incrementalPrAnalysis_prWithoutChanges_producesUnchangedFilesWithAllFiles() throws IOException {
+    var projectDir = TestUtils.projectDir(temp, "IncrementalPRAnalysis");
+    var logWriter = new StringWriter();
+    StreamConsumer.Pipe logsConsumer = new StreamConsumer.Pipe(logWriter);
+
+    // Initial build - master
+    runBeginStep(projectDir, logsConsumer);
+    runBuild(projectDir, logsConsumer);
+    runEndStep(projectDir, logsConsumer);
+    var logs = logWriter.toString();
+    assertThat(logs).doesNotContain("Incremental PR analysis: 3 files out of 3 are unchanged.");
```

Comment:
Is this a useful check? 
This is not possible to be raised on master, right?
Would it not be better if we checked the same message you check on the master-only test:
```java
assertThat(logWriter.toString()).contains(
      "Processing analysis cache",
      "Incremental PR analysis: Base branch parameter was not provided.",
      "Cache data is empty. A full analysis will be performed.");
```

Reply:
I've added the check to ensure and point out that the message was logged by the second analysis since the logs are common. I wouldn't add the same check again (since this is verified separately) but I'm considering now to use separate logs and simplify the code a bit.

Reply:
I actually missed that you use the same logs, my bad. :)
Maybe separate logs would indeed be better for readability.
But now it makes more sense why you do the `not-contains` check, thanks!